spec/lib/configgin_spec.rb

```diff
@@ -76,70 +75,51 @@
+      before(:each) {
+        stub_const('ENV',
+                   # exported properties secret is only created for the main instance group container
+                   'KUBERNETES_CONTAINER_NAME' => 'instance-group',
+                   'CONFIGGIN_VERSION_TAG' => '1.2.3')
       }
-    }
+      it 'writes initial digest values to secret' do
+        # Tag does not exist in secret, so create both exported properties setting *and* initial
+        # digest values. Do *not* create any annotation on the statefulset.
+        subject.run
+        secret = client.get_secret('instance-group', 'the-namespace')
+        expect(secret).not_to be_nil
+        exported_properties = Base64.decode64(secret.data['skiff-exported-properties-loggregator_agent'])
+        initial_digest = Base64.decode64(secret.data['skiff-initial-digest-loggregator_agent'])
+        expect(initial_digest).to eq property_digest(JSON.parse(exported_properties))
+
+        statefulset = client.get_stateful_set('debugger', 'the-namespace')
+        expect(statefulset).not_to be_nil
+        expect(statefulset.spec.template.metadata.annotations['skiff-in-props-instance-group-loggregator_agent']).to be_nil
+      end
 
-    it 'patches the jobs' do
-      subject.patch_job_metadata(jobs)
-      pod = client.get_pod('pod-0', 'the-namespace')
-      expect(pod).not_to be_nil
-      secret = client.get_secret("#{pod.metadata.name}-#{pod.metadata.uid}")
-      expect(secret).not_to be_nil
+      it 'patches the affected statefulset' do
+        # Create secret with tag value, so initial_digest should *not* be written to the secret,
+        # but affected statefulsets should be annotated with the expected digest value.
+        secret = Kubeclient::Resource.new
+        secret.metadata = {
+          name: 'instance-group',
+          namespace: 'the-namespace',
+        }
+        secret.data = { ENV['CONFIGGIN_VERSION_TAG'] => "" }
+        client.create_secret(secret)
 
-      jobs.each do |name, job|
-        property_name = "skiff-exported-properties-#{name}"
-        expect(Base64.decode64(secret.data[property_name])).to eq job.exported_properties.to_json
-        digest_name = "skiff-exported-digest-#{name}"
-        expect(Base64.decode64(secret.data[digest_name])).to eq property_digest(job.exported_properties)
+        subject.run
+        secret = client.get_secret('instance-group', 'the-namespace')
+        expect(secret).not_to be_nil
+        expect(secret.data['skiff-initial-digest-loggregator_agent']).to be_nil
+        exported_properties = Base64.decode64(secret.data['skiff-exported-properties-loggregator_agent'])
+
+        statefulset = client.get_stateful_set('debugger', 'the-namespace')
+        expect(statefulset).not_to be_nil
+        expected_digest = statefulset.spec.template.metadata.annotations['skiff-in-props-instance-group-loggregator_agent']
+        expect(expected_digest).to eq property_digest(JSON.parse(exported_properties))
```

Comment:
The naming's backwards here, no? (The digest we fetched is the actual digest, not the expected.)